packages/dcos-integration-test/extra/test_dcos_log.py

```diff
@@ -236,3 +236,104 @@ def get_task_url(dcos_api_session, task_name, stream=False):
     return '/system/v1/agent/{}/logs/v1/{}/framework/{}/executor/{}/container/{}'.format(slave_id, endpoint_type,
                                                                                          framework_id, executor_id,
                                                                                          container_id)
+
+
+def test_log_v2_text(dcos_api_session):
+    for node in dcos_api_session.masters + dcos_api_session.all_slaves:
+        response = dcos_api_session.logs.get('v2/component?limit=10', node=node)
+        check_response_ok(response, {'Content-Type': 'text/plain'})
+
+        # expect 10 lines
+        lines = list(filter(lambda x: x != '', response.content.decode().split('\n')))
+        assert len(lines) == 10, 'Expect 10 log entries. Got {}. All lines {}'.format(len(lines), lines)
+
+
+def test_log_v2_server_sent_events(dcos_api_session):
+    for node in dcos_api_session.masters + dcos_api_session.all_slaves:
+        response = dcos_api_session.logs.get('v2/component?limit=1', node=node, headers={'Accept': 'text/event-stream'},
+                                             stream=True)
+        check_response_ok(response, {'Content-Type': 'text/event-stream', 'Cache-Control': 'no-cache'})
+        lines = response.iter_lines()
+        sse_id = next(lines)
+        assert sse_id, 'First line must be id. Got {}'.format(sse_id)
+        data = next(lines).decode('utf-8', 'ignore')
+        validate_sse_entry(data)
+
+
+def test_log_v2_stream(dcos_api_session):
+    for node in dcos_api_session.masters + dcos_api_session.all_slaves:
+        response = dcos_api_session.logs.get('v2/component?skip=-1', node=node, stream=True,
+                                             headers={'Accept': 'text/event-stream'})
+        check_response_ok(response, {'Content-Type': 'text/event-stream', 'Cache-Control': 'no-cache'})
+        lines = response.iter_lines()
+        sse_id = next(lines)
+        assert sse_id, 'First line must be id. Got {}'.format(sse_id)
+        data = next(lines).decode('utf-8', 'ignore')
+        validate_sse_entry(data)
+
+
+def test_log_v2_proxy(dcos_api_session):
```

Comment:
Some documentation about what `proxy` means here would be good. IIRC the master will send a redirect to the client in the case that you're querying for agent logs right? `proxy` suggests that it is not sending the redirect.

Reply:
right, the discover endpoint is being checked in the other tests, for example `dcos_api_session.logs.get('v2/task/sleep1')` will go to discover endpoint and will be redirected to the appropriate agent. The purpose of this test is to validate that master adminrouter is proxying to the agents correctly.